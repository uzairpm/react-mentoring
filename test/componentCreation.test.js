import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Es6Component from '../src/components/es6-component';
import FunctionComponent from '../src/components/functional-component';

beforeAll(() => {
    configure({ adapter: new Adapter() });
});
describe('Check Component creation', () => {
    test('ES6 component', () => {
        const element = shallow(<Es6Component />);
        expect(element).toMatchSnapshot();
        expect(element.text()).toEqual('Hello World using ES6 React.Component');
    });
    test('Functional component', () => {
        const element = shallow(<FunctionComponent msg="bye"/>);
        expect(element.text()).toEqual('Stateless functional component says bye');
    });
});
