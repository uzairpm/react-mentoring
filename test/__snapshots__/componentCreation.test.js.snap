// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Component creation ES6 component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Es6Component />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Hello World using ES6 React.Component",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Hello World using ES6 React.Component",
    "type": "h2",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello World using ES6 React.Component",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Hello World using ES6 React.Component",
      "type": "h2",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
